pkg/virtctl/expose/expose.go

```diff
@@ -59,262 +61,232 @@ Possible types are (case insensitive, both single and plurant forms):
+	service := &k8sv1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      e.serviceName,
+			Namespace: e.namespace,
 		},
-		Spec: v1.ServiceSpec{
-			Ports:          ports,
-			Selector:       serviceSelector,
-			ClusterIP:      clusterIP,
-			Type:           serviceType,
-			LoadBalancerIP: loadBalancerIP,
-			IPFamilies:     ipFamilies,
+		Spec: k8sv1.ServiceSpec{
+			Ports:      ports,
+			Selector:   serviceSelector,
+			ClusterIP:  e.clusterIP,
+			Type:       e.serviceType,
+			IPFamilies: e.ipFamilies,
 		},
 	}
-
-	// set external IP if provided
-	if len(externalIP) > 0 {
-		service.Spec.ExternalIPs = []string{externalIP}
+	if len(e.externalIP) > 0 {
+		service.Spec.ExternalIPs = []string{e.externalIP}
 	}
-
-	if ipFamilyPolicy != "" {
-		service.Spec.IPFamilyPolicy = &ipFamilyPolicy
+	if e.ipFamilyPolicy != "" {
+		service.Spec.IPFamilyPolicy = &e.ipFamilyPolicy
 	}
-
-	major, minor, err := serverVersion(virtClient)
-	if err != nil {
-		return err
+	if _, err := e.client.CoreV1().Services(e.namespace).Create(context.Background(), service, metav1.CreateOptions{}); err != nil {
+		return fmt.Errorf("service creation failed: %v", err)
 	}
 
-	if major > 1 || (major == 1 && minor >= 20) {
-		_, err = virtClient.CoreV1().Services(namespace).Create(context.Background(), service, k8smetav1.CreateOptions{})
-		if err != nil {
-			return fmt.Errorf("service creation failed for k8s >= 1.20: %v", err)
-		}
-		// For k8s < 1.20 we have to "migrate" the "ipFamilies" field to
-		// "ipFamily" we do this using an unstructured approach
-	} else {
-		if len(ipFamilies) > 1 {
-			return fmt.Errorf("k8s < 1.20 doesn't support multiple ip families")
-		}
-
-		if ipFamilyPolicy != "" {
-			return fmt.Errorf("k8s < 1.20 doesn't support 'ipFamilyPolicy'")
-		}
-		// convert the Service to unstructured.Unstructured
-		unstructuredService, err := runtime.DefaultUnstructuredConverter.ToUnstructured(service)
-		if err != nil {
-			return err
-		}
+	return nil
+}
 
-		// Add ipFamily field with proper content
-		err = unstructured.SetNestedField(unstructuredService, string(ipFamilies[0]), "spec", "ipFamily")
-		if err != nil {
-			return err
-		}
+func convertProtocol(strProtocol string) (k8sv1.Protocol, error) {
+	switch strings.ToLower(strProtocol) {
+	case "tcp":
+		return k8sv1.ProtocolTCP, nil
+	case "udp":
+		return k8sv1.ProtocolUDP, nil
+	default:
+		return "", fmt.Errorf("unknown protocol: %s", strProtocol)
+	}
+}
 
-		// try to create the service on the cluster
-		_, err = virtClient.DynamicClient().Resource(schema.GroupVersionResource{Version: "v1", Resource: "services"}).Namespace(namespace).Create(context.Background(), &unstructured.Unstructured{Object: unstructuredService}, k8smetav1.CreateOptions{})
-		if err != nil {
-			return fmt.Errorf("service creation failed for k8s < 1.20: %v", err)
-		}
+func convertServiceType(strServiceType string) (k8sv1.ServiceType, error) {
+	switch strings.ToLower(strServiceType) {
+	case "clusterip":
```

Comment:
Are there no public constants for these?

Reply:
Addressed to use constants.